usermanager/src/test/java/org/pesmypetcare/usermanager/clients/pet/PetCollectionsManagerClientTest.java

```diff
@@ -0,0 +1,644 @@
+                + "}"
+        );
+        washData = new WashData("It smelt so bad", 54);
+        washCollectionList = new ArrayList<>();
+        washCollectionList.add(new Wash(DATE_1, washData));
+        washCollectionList.add(new Wash(DATE_2, washData));
+        washCollectionList.add(new Wash(DATE_3, washData));
+
+        weightCollectionJson = new StringBuilder("[\n"
+                + "  {\n"
+                + "    \"body\": {\n"
+                + "      \"value\": 54\n"
+                + "    },\n"
+                + "    \"key\": \"1990-01-08T15:20:30\"\n"
+                + "  },{\n"
+                + "    \"body\": {\n"
+                + "      \"value\": 54\n"
+                + "    },\n"
+                + "    \"key\": \"1995-01-08T15:20:30\"\n"
+                + "  },{\n"
+                + "    \"body\": {\n"
+                + "      \"value\": 54\n"
+                + "    },\n"
+                + "    \"key\": \"1998-01-08T15:20:30\"\n"
+                + "  }\n"
+                + "]");
+        weightDataJson = new StringBuilder("{\n"
+                + " \"value\": 54\n"
+                + "}"
+        );
+        weightData = new WeightData(54);
+        weightCollectionList = new ArrayList<>();
+        weightCollectionList.add(new Weight(DATE_1, weightData));
+        weightCollectionList.add(new Weight(DATE_2, weightData));
+        weightCollectionList.add(new Weight(DATE_3, weightData));
+    }
+
+    @Test
+    public void getAllExercises() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(exerciseCollectionJson);
+        List<Exercise> response = client.getAllExercises(ACCESS_TOKEN, USERNAME, PET_NAME);
+        assertEquals("Should return a list", exerciseCollectionList, response);
+    }
+
+    @Test
+    public void getExercisesBetween() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(exerciseCollectionJson);
+        List<Exercise> response = client.getExercisesBetween(ACCESS_TOKEN, USERNAME, PET_NAME, DATE_1, DATE_3);
+        assertEquals("Should return a list", exerciseCollectionList, response);
+    }
+
+    @Test
+    public void getExercises() throws ExecutionException, InterruptedException {
+        given(taskManager.resetTaskManager()).willReturn(taskManager);
+        given(taskManager.execute(anyString(), anyString())).willReturn(taskManager);
+        given(taskManager.get()).willReturn(exerciseDataJson);
+        ExerciseData response = client.getExercise(ACCESS_TOKEN, USERNAME, PET_NAME, DATE_1);
+        assertEquals("Should return the specified element", exerciseData, response);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Should return the specified element" appears 9 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5549434)